File: xadapterLibrary/src/main/java/com/xadapter/widget/SimpleLoadMore.java
package com.xadapter.widget;

import android.content.Context;
import android.graphics.drawable.AnimationDrawable;
import android.support.annotation.Nullable;
import android.support.v7.widget.AppCompatImageView;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.View;

import com.xadapter.R;

/**
 * by y on 2016/9/29
 */

public class SimpleLoadMore extends XLoadMoreView {

    private AppCompatImageView progressBar;
    private AppCompatTextView mText;
    private AnimationDrawable animationDrawable;

    public SimpleLoadMore(Context context) {
        super(context);
    }

    public SimpleLoadMore(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public SimpleLoadMore(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    @Override
    protected int getLayoutId() {
        return R.layout.simple_load_more;
    }

    @Override
    public void initView() {
        progressBar = (AppCompatImageView) findViewById(R.id.progressbar);
        mText = (AppCompatTextView) findViewById(R.id.tv_tips);
        animationDrawable = (AnimationDrawable) progressBar.getBackground();
        mText.setText("上拉加载");
    }

    @Override
    protected void onStart() {
        progressBar.setVisibility(GONE);
        animationDrawable.stop();
    }

    @Override
    protected void onLoad() {
        progressBar.setVisibility(View.VISIBLE);
        animationDrawable.start();
        mText.setText("正在加载...");
    }

    @Override
    protected void onNoMore() {
        progressBar.setVisibility(View.GONE);
        animationDrawable.stop();
        mText.setText("没有数据了");
    }

    @Override
    protected void onSuccess() {
        progressBar.setVisibility(View.GONE);
        animationDrawable.stop();
        mText.setText("加载成功");
    }

    @Override
    protected void onError() {
        progressBar.setVisibility(View.GONE);
        animationDrawable.stop();
        mText.setText("加载失败");
    }

    @Override
    protected void onNormal() {
        progressBar.setVisibility(View.GONE);
        animationDrawable.stop();
        mText.setText("上拉加载");
    }
}

File: xadapterLibrary/src/main/java/com/xadapter/simple/SimpleLoadMore.kt
package com.xadapter.simple

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.AnimationDrawable
import android.view.View
import com.xadapter.R
import com.xadapter.XLoadMoreView
import kotlinx.android.synthetic.main.simple_load_more.view.*

/**
 * by y on 2016/9/29
 */

class SimpleLoadMore(context: Context) : XLoadMoreView(context, R.layout.simple_load_more) {

    private lateinit var animationDrawable: AnimationDrawable

    public override fun initView() {
        animationDrawable = progressbar.background as AnimationDrawable
        tv_tips.setTextColor(Color.BLACK)
        progressbar.visibility = View.GONE
        tv_tips.text = "上拉加载"
    }

    override fun onStart() {
        progressbar.visibility = View.GONE
        animationDrawable.stop()
    }

    override fun onLoad() {
        progressbar.visibility = View.VISIBLE
        animationDrawable.start()
        tv_tips.text = "正在加载..."
    }

    override fun onNoMore() {
        progressbar.visibility = View.GONE
        animationDrawable.stop()
        tv_tips.text = "没有数据了"
    }

    override fun onSuccess() {
        progressbar.visibility = View.GONE
        animationDrawable.stop()
        tv_tips.text = "加载成功"
    }

    override fun onError() {
        progressbar.visibility = View.GONE
        animationDrawable.stop()
        tv_tips.text = "加载失败"
    }

    override fun onNormal() {
        progressbar.visibility = View.GONE
        animationDrawable.stop()
        tv_tips.text = "上拉加载"
    }
}
